opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1362,73 +1362,117 @@ template cpgrid::Entity<3> Dune::createEntity(const CpGrid&, int, bool);
+    std::vector<int> all_patch_corners; 
+    std::vector<int> all_patch_faces; 
+    std::vector<int> all_patch_cells;
+    std::vector<std::vector<std::array<int,2>>> all_patch_boundary_old_to_new_corners;
+    std::vector<std::vector<std::tuple<int,std::vector<int>>>> all_patch_boundary_old_to_new_faces;
+    std::vector<std::vector<std::tuple<int,std::vector<int>>>> all_patch_parent_to_children_cells;
+    std::vector<std::vector<std::array<int,2>>> all_patch_child_to_parent_cells;
+    all_patch_boundary_old_to_new_corners.reserve(num_patches);
+    all_patch_boundary_old_to_new_faces.reserve(num_patches);
+    all_patch_parent_to_children_cells.resize(num_patches);
+    all_patch_child_to_parent_cells.resize(num_patches);
+    // For level0, attach children to each parent cell. For no parents, entry {-1, {-1}} representing {no level, {no children}}
     auto& l0_parent_to_children_cells = (*data_[0]).parent_to_children_cells_;
-    const std::vector<int>& no_child = {-1};
-    const int& no_level = -1;
-    const std::tuple<int, std::vector<int>>& no_children = std::make_tuple(no_level, no_child);
-    // For cells with no children, we set {-1,{-1}}. Entries of actual parents will be rewritten.
     l0_parent_to_children_cells.resize(data_[0]-> size(0));
-    for (int cell = 0; cell < data_[0] -> size(0); ++cell){
-        l0_parent_to_children_cells[cell] = no_children;
-    }
-    // For level1/LGR, attach to each child-cell its parent.
-    std::vector<std::array<int,2>>& l1_child_to_parent_cells = (*data_[1]).child_to_parent_cells_;
-    // For cells with no parent, we set {-1,-1}. Entries with actual parents will be rewritten.
-    const std::array<int,2>& no_parent = {-1,-1};
-    l1_child_to_parent_cells.resize(data_[1]-> size(0));
-    for (int cell = 0; cell < data_[1]->size(0); ++cell){
-        l1_child_to_parent_cells[cell] = no_parent;
+    const std::vector<int> no_children = {-1};    
+    const std::tuple<int,std::vector<int>> has_no_children = std::make_tuple(-1, no_children);
+    for (int cell = 0; cell < data_[0]-> size(0); ++cell){
+        l0_parent_to_children_cells[cell] = has_no_children;
     }
-    // Rewrite entries for actual parents in level0 and actual children in level1.
-    for (const auto& [parent, children] : parent_to_children_cells) {
-        l0_parent_to_children_cells[parent] = {1, children}; // {level LGR, {child0, child1, ...}}
-        for (const auto& child : children){
-            l1_child_to_parent_cells[child] = {0, parent}; // {level of parent cell, parent cell index in that level}
+    // Get patches corner, face, and cell indices.
+    for (int patch = 0; patch < num_patches; ++patch){
+        const auto& [corners, faces, cells] = (*(this->data_[0])).getPatchGeomIndices(startIJK_vec[patch], endIJK_vec[patch]);
+        all_patch_corners.reserve(corners.size());
+        for (const auto& corner : corners){
+            all_patch_corners.push_back(corner);
         }
```

Comment:
For the second patch the reserve call probably has little effect as the vector is not empty anymore. 

If that would be fixed then I think this does the same as if you would use the [insert](https://en.cppreference.com/w/cpp/container/vector/insert) function of std::vector `all_patch_corners.insert(all_patch_corners.end(),corners.begin(), corners.end());`. Maybe it is better to use that one)?

Similar for other for face, etc below.

Reply:
I did the following modification:

`all_patch_corners.reserve(all_patch_corners.size() + corners.size());
        for (const auto& corner : corners){
            all_patch_corners.push_back(corner);
        }`
To enlarge the capacity of the vector in each step.